core/pom.xml

```diff
@@ -370,5 +370,28 @@
             </activation>
             <!-- not including license-maven-plugin is sufficent to expose default license -->
         </profile>
+        <profile>
+            <id>release</id>
+            <build>
+                <plugins>
+                    <plugin>
+                        <groupId>org.apache.maven.plugins</groupId>
+                        <artifactId>maven-javadoc-plugin</artifactId>
+                        <executions>
+                            <execution>
+                                <id>attach-javadocs</id>
+                                <goals>
+                                    <goal>jar</goal>
+                                </goals>
+                                <configuration>
+                                    <additionalparam>-Xdoclint:none</additionalparam>
```

Comment:
I took a stab at fixing the doclint errors but quite frankly, it's just too many right now to fix in the scope of this PR. Some of them can be fixed mechanically (e. g. with `sed`) but many more require manual intervention.

Take a look at https://github.com/joschi/elasticsearch/tree/publish-javadoc-artifact-including-javadoc-fixes for some of the fixes.